src/vs/editor/browser/widget/diffEditorWidget.ts

```diff
@@ -249,11 +250,16 @@ export class DiffEditorWidget extends Disposable implements editorBrowser.IDiffE
 		@IThemeService themeService: IThemeService,
 		@INotificationService notificationService: INotificationService,
 		@IContextMenuService contextMenuService: IContextMenuService,
-		@IEditorProgressService private readonly _editorProgressService: IEditorProgressService
+		@IEditorProgressService private readonly _editorProgressService: IEditorProgressService,
+		@IConfigurationService private readonly _configurationService: IConfigurationService
 	) {
 		super();
 		codeEditorService.willCreateDiffEditor();
-
+		this._register(this._configurationService.onDidChangeConfiguration(e => {
```

Comment:
The DiffEditorWidget should not listen on the configuration service directly. The embedder of the DiffEditorWidget needs to do this and forward the settings using `diffEditor.updateOptions()`. The reason is that the editor and diff editor are usable as simple widgets and can thus be disconnected from the workbench settings.